__tests__/users.test.ts

```diff
@@ -14,19 +14,215 @@ beforeAll(async () => {
+        phoneNumber: "1234567890",
+      },
+      {
+        id: "6",
+        firstName: "Admin",
+        lastName: "Doe",
+        email: "admin2@gmail.com",
+        role: "Admin",
+        address: "123 Main Street",
+        phoneNumber: "1234567890",
+      },
+    ];
+
+    const res = await makeHTTPRequest(
+      userHandler,
+      "/api/users/",
+      undefined,
+      "GET",
+      undefined,
+      StatusCodes.OK,
+      undefined
+    );
+
+    const returnedUsers = JSON.parse(res._getData());
+
+    expect(expected.length).toBe(returnedUsers.length);
+    expect(returnedUsers).toEqual(expect.arrayContaining(returnedUsers));
+  });
+
+  test("look for all teachers", async () => {
+    const expected: User[] = [
+      {
+        id: "2",
+        firstName: "Teacher",
+        lastName: "Doe",
+        email: "teacher@gmail.com",
+        role: "Teacher",
+        address: "123 Main Street",
+        phoneNumber: "1234567890",
+      },
+      {
+        id: "5",
+        firstName: "Teacher",
+        lastName: "Doe",
+        email: "teacher2@gmail.com",
+        role: "Teacher",
+        address: "123 Main Street",
+        phoneNumber: "1234567890",
+      },
+    ];
+
+    const query = {
+      role: "Teacher",
+    };
+
+    await makeHTTPRequest(
+      userHandler,
+      "/api/users/",
+      query,
+      "GET",
+      undefined,
+      StatusCodes.OK,
+      expected
+    );
+  });
+
+  test("look for all students", async () => {
+    const expected: User[] = [
+      {
+        id: "1",
+        firstName: "John",
+        lastName: "Doe",
+        email: "john@gmail.com",
+        role: "Student",
+        address: "123 Main Street",
+        phoneNumber: "1234567890",
+      },
+      {
+        id: "4",
+        firstName: "John",
+        lastName: "Doe",
+        email: "john2@gmail.com",
+        role: "Student",
+        address: "123 Main Street",
+        phoneNumber: "1234567890",
+      },
+    ];
+
+    const query = {
+      role: "Student",
+    };
+
+    await makeHTTPRequest(
+      userHandler,
+      "/api/users/",
+      query,
+      "GET",
+      undefined,
+      StatusCodes.OK,
+      expected
+    );
+  });
+
+  test("look for all admin", async () => {
+    const expected: User[] = [
+      {
+        id: "3",
+        firstName: "Admin",
+        lastName: "Doe",
+        email: "admin@gmail.com",
+        role: "Admin",
+        address: "123 Main Street",
+        phoneNumber: "1234567890",
+      },
+      {
+        id: "6",
+        firstName: "Admin",
+        lastName: "Doe",
+        email: "admin2@gmail.com",
+        role: "Admin",
+        address: "123 Main Street",
+        phoneNumber: "1234567890",
+      },
+    ];
+
+    const query = {
+      role: "Admin",
+    };
+
+    await makeHTTPRequest(
+      userHandler,
+      "/api/users/",
+      query,
+      "GET",
+      undefined,
+      StatusCodes.OK,
+      expected
+    );
+  });
+});
+
```

Comment:
Add a test case where the value of the role query param is an invalid Role? Although I'm not sure what you want the expected behavior to be in this case, I made another comment about this in pages/api/users/index.ts